azkaban-common/src/main/java/azkaban/project/Dependency.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.project;
+
+import azkaban.Constants;
+import azkaban.utils.Props;
+import com.google.common.base.Preconditions;
+
+/**
+ * Dependency is an immutable class which holds
+ * all the data and properties of a dependency.
+ */
+public class Dependency {
+
```

Comment:
Why don't we also have name and type fields for Dependency? Would that be more convenient than getting it from props every time?

Reply:
My thinking is to use fewer class member variables for simplicity. Given props already serves the purpose of holding properties, i don't think we need special class member variables for name and type for now.